src/gxcore/common/geolocation.ts

```diff
@@ -0,0 +1,241 @@
+import { notImplemented } from "../../misc/helpers";
+
+export class GeneXusCommonGeolocation {
+
+  /**
+   * Returns the current location for the device
+   * @param minAccuracy
+   * @param timeout
+   * @param includeHeadingAndSpeed
+   * @param ignoreErrors
+   * @return any
+   */
+  public static getMyLocation(minAccuracy: number, timeout: number, includeHeadingAndSpeed:boolean, ignoreErrors:boolean): Promise<GeneXusCommonGeolocationInfoData> {
+    return new Promise<GeneXusCommonGeolocationInfoData>((resolve, reject) => {
+      if (navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition((position: any) => { 
+          let geoPosition = new GeneXusCommonGeolocationInfoData();
+          geoPosition.Location = position.coords.latitude.toString() + "," + position.coords.longitude.toString();
+          geoPosition.Heading = position.coords.heading;
+          geoPosition.Precision = position.coords.accuracy;
+          geoPosition.Description = "";
+          geoPosition.Speed = position.coords.speed;
+          geoPosition.Time = position.timestamp;
+          resolve(geoPosition);
+        }, 
+        (error) => {
+          if (ignoreErrors) { 
+            let geoPosition = new GeneXusCommonGeolocationInfoData();
+            geoPosition.Location = "";
+            geoPosition.Heading = 0;
+            geoPosition.Precision = 0;
+            geoPosition.Description = "";
+            geoPosition.Speed = 0;
```

Comment:
This code is unnecessary, that's what the `GeneXusCommonGeolocationInfoData` constructor does (except for the Time field which I left outside of this comment...)